### DISCUSSION TITLE
Error while using Morphometric calculator

### DISCUSSION THREAD

**Discussion by MartinaFrid** [UMEP-dev/UMEP#486] [Q&A]
Hi 

I am running into an error while using the Morphometric calculator. I am using building heights and a grid shape file. The grid is a bit tilted in relation to the building height pixels and cuts through the pixels. Does the grid need to align for this to work or is there a different way to approach this? 

The error I get is:

LINE 92 "c[int(i)-1, :] = np.sum((filt*a[int(i)-1:i+1, :]), 0)" 
ERROR MESSAGE: operands could not be broadcast together with shapes (2,1128) (2,704) 

 LINE 237 "zH = immorphresult["zH"]" 
ERROR MESSAGE: 'NoneType' object is not subscriptable

QGIS version: 3.28.1
UMEP version: 3.20.9

Thank you in advance

The data is here:
https://www.dropbox.com/scl/fo/9nosyoyorljj1nss4imyb/h?dl=0&rlkey=yzm63hvheynx8ub13xit0rqhs

**Comment by biglimp** [comment]
First of all, I think you found a bug that we need to fix. I assume you use the search option "Search throughout the grid extent", yes?

Then I would reproject all your data so that your grids got a squared shape (e.g. EPSG:4326). This tool cannot really deal with rectangular shapes as you need a square raster to rotate for the different wind directions.  

As a work around for now, you can Serach from grid centriod to make it work.

![image](https://user-images.githubusercontent.com/11908356/208461917-bc09a47c-c7fb-4aaa-9c26-35132d3e1d32.png)

**Comment by MartinaFrid** [comment]
Hi 

Yes, I am using "Search through the grid extent". 

Okay that make sense. Thank you for the clarification on for the help.

**Comment by biglimp** [comment]
Hm, Now I tried with another dataset and I worked for me. Maybe there is something strange with the CRS you are using. Could you try change and see if that solved your issue? Maybe use 27700 for example.

**Reply by biglimp** [reply]
Hold on, I though you were in London. Are your data in GErmany or something? Dont use 27700 then.

**Reply by MartinaFrid** [reply]
Yes, this is over Berlin

**Reply by biglimp** [reply]
Ok, I find the misstake. It is not an easy fix so reproject your layers into 4326 and then it will work.

**Reply by MartinaFrid** [reply]
I still get the same error after reprojecting the layers to 4326. Did it work for you? I will try with a different data set and see if it works. Sorry for all the trouble

**Reply by biglimp** [reply]
Worked for me with your data. You need to reproject both the vector grid (Reproject vector layer) and the raster (using Warp). Both are found in the Processing Toolbox in QGIS.

**Reply by MartinaFrid** [reply]
Thank you
